src/status_im/ui/screens/network_settings/events.cljs

```diff
@@ -0,0 +1,33 @@
+(ns status-im.ui.screens.network-settings.events
+  (:require [re-frame.core :refer [dispatch dispatch-sync after] :as re-frame]
+            [status-im.utils.handlers :refer [register-handler] :as handlers]
+            [status-im.data-store.networks :as networks]
+            [status-im.ui.screens.network-settings.navigation]))
+
+;;;; FX
+
+(re-frame/reg-fx
+  ::save-networks
+  (fn [networks]
+    (networks/save-all networks)))
+
+;; handlers
+
+(handlers/register-handler-fx
+  :add-networks
+  (fn [{{:networks/keys [networks] :as db} :db} [_ new-networks]]
+    (let [identities    (set (keys networks))
+          new-networks' (->> new-networks
```

Comment:
Put in separate fn? Seems to do some kind of clean up but not immediately clear what